crates/uv/src/commands/project/add.rs

```diff
@@ -33,10 +40,93 @@ pub(crate) async fn add(
     // Find the project requirements.
     let project = ProjectWorkspace::discover(&std::env::current_dir()?, None).await?;
 
+    // Discover or create the virtual environment.
+    let venv = project::init_environment(project.workspace(), python.as_deref(), cache, printer)?;
```

Comment:
Note we should only discover an interpreter not create an environment if syncing is disabled (looks like there's no toggle for that yet)